### PR TITLE
Os 441 rpc kameran ajot

### PR DESCRIPTION
**Pull request by Somac**
https://jira.nls.fi/browse/OS-441

### CODE REVIEW

File: bundles/mapping/mapmodule/MapModuleClass.ol.js [hunk 1]
```diff
@@ -538,24 +599,39 @@ export class MapModule extends AbstractMapModule {
...
+                duration: 3000,
+                easing: easeOut
+            });
+        } else {
+            this._animateTo(lonlat, zoomN, options.animation, options.duration);
+        }
+
         this.updateDomain();
         if (suppressEnd !== true) {
             this.notifyMoveEnd();
```
Comment (ZakarFin) [inline bundles/mapping/mapmodule/MapModuleClass.ol.js]: With this implementation the map moved event comes at the beginning of the animation. We would need to trigger it at the end.

Comment (ZakarFin) [reply bundles/mapping/mapmodule/MapModuleClass.ol.js]: Also I think we need to have centerMap() to skip any animation by default as users might expect the map to be on a given location after this is called.

File: bundles/mapping/mapmodule/MapModuleClass.ol.js [hunk 2]
```diff
@@ -538,24 +599,39 @@ export class MapModule extends AbstractMapModule {
...
         if (!this.isValidLonLat(lonlat.lon, lonlat.lat)) {
             return false;
         }
-        this.getMap().getView().setCenter([lonlat.lon, lonlat.lat]);
         if (zoom === null || zoom === undefined) {
             zoom = this.getMapZoom();
         }
-        this.getMap().getView().setZoom(zoom);
+
+        const zoomN = zoom.type === 'scale' ? view.getZoomForResolution(zoom.value) : zoom.value;
```
Comment (ZakarFin) [inline bundles/mapping/mapmodule/MapModuleClass.ol.js]: We need to check if zoom was a number and treat it as zoom.value and use this logic if zoom was an object to be backwards compatible.

**Comment by jheijari** [comment]
Closing as a duplicate. See https://github.com/oskariorg/oskari-frontend/pull/981.

**Pull request closed without merging**